code/modules/mob/living/simple_animal/friendly/rat.dm

```diff
@@ -0,0 +1,169 @@
+/mob/living/simple_animal/rat
+	name = "rat"
+	real_name = "rat"
+	desc = "It's a big, disease-ridden rodent."
+	icon_state = "rat_gray"
+	icon_living = "rat_gray"
+	icon_dead = "rat_gray_dead"
+	speak = list("Squeek!","SQUEEK!","Squeek?")
+	speak_emote = list("squeeks","squeeks","squiks")
+	emote_hear = list("squeeks","squeaks","squiks")
+	emote_see = list("runs in a circle", "shakes", "scritches at something")
+	mob_size = MOB_SIZE_SMALL
+	speak_chance = 1
+	turns_per_move = 5
+	see_in_dark = 6
+	maxHealth = 10
+	health = 10
+	meat_type = /obj/item/reagent_container/food/snacks/meat
+	response_help  = "pets the"
+	response_disarm = "gently pushes aside the"
+	response_harm   = "stamps on the"
+	density = FALSE
+	layer = ABOVE_LYING_MOB_LAYER
+	min_oxy = 16 //Require atleast 16kPA oxygen
+	minbodytemp = 223 //Below -50 Degrees Celcius
+	maxbodytemp = 323 //Above 50 Degrees Celcius
+	universal_speak = FALSE
+	universal_understand = TRUE
+	holder_type = /obj/item/holder/rat
+	var/body_color //black, brown, gray and white, leave blank for random
+
+/mob/living/simple_animal/rat/Life(delta_time)
+	..()
+	if(!stat && prob(speak_chance))
+		for(var/mob/cur_mob in view())
+			cur_mob << 'sound/effects/mousesqueek.ogg'
```

Comment:
This should be amended to use `FOR_DVIEW` like done here https://github.com/cmss13-devs/cmss13/pull/6712/files#diff-75492799845accd06464d0ab1acd1f45b5dae4037bb12060c2d4f76f91e87dc4
OR rats should inherit from `/mob/living/simple_animal/mouse` e.g. `/mob/living/simple_animal/mouse/rat` to reuse all behaviors (for the most part you'd just need to make a `var/base_icon = "mouse"` or `"rat"` and replace any icon_state stuff from "mouse_whatever" and "rat_whatever" to "[base_icon]_whatever".

Reply:
Ah man I didn't see these until now, could you help me out with these please? I'm afraid of messing things up.